app/models/post.rb

```diff
@@ -1,3 +1,28 @@
+# == Schema Information
+#
+# Table name: posts
+#
+#  id               :bigint           not null, primary key
+#  title            :string
+#  text             :text
+#  created_at       :datetime         not null
+#  updated_at       :datetime         not null
+#  author_id        :bigint           not null
+#  likes_counter    :integer
+#  comments_counter :integer
+#
 class Post < ApplicationRecord
-  belogns_to :user
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id', counter_cache: true
+  has_many :comments
+  has_many :likes
+
+  # updates the posts counter for a user
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
+
+  # returns the 5 most recent comments for a given post.
+  def show_recent_comments
+    comments.where(post_id: id).first(5)
+  end
```

Comment:
**OPTIONAL**

- [ ] It will be nice to use the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_posts_counter` method every time a new post has been saved to the database.

- [ ] Also make the `update_posts_counter` method private.